### PR TITLE
add polygonal mesh support to blueprint

### PR DESCRIPTION
**Pull request by xjrc**
The changes in this pull request add support for polygonal mesh representation in Blueprint. This includes rudimentary functionality such as updating the Blueprint verification code to accept polygonal meshes, but extends to also include helper functions to help users convert to and work with this new representation. The full list of updates that comprise this pull request is provided below:

- [x] Update `blueprint::mesh::verify` to support polygonal mesh data.
- [X] Implement `blueprint::mesh::unstructured::to_polygonal` to allow trivial conversions from an implicit topology to a polygonal equivalent.
- [X] Implement `blueprint::mesh::unstructured::generate_offsets` to simplify the process of generating element offsets for polygonal mesh data.
- [X] Update the documentation to include an example of a polygonal mesh.
- [x] Add test cases for the `blueprint::mesh::verify` changes.
- [X] Add test cases for the `blueprint::mesh::unstructured::to_polygonal` function.
- [X] Add test cases for the `blueprint::mesh::unstructured::generate_offsets` function.
- [x] Implement `blueprint::mesh::examples::polytess` to demo heterogeneous polygonal meshes.
- [x] Add test cases to validate the new `blueprint::mesh::examples::polytess` mesh example.
- [x] Add documentation for the new `blueprint::mesh::examples::polytess` mesh example.

### CODE REVIEW

**Comment by xjrc** [comment]
@cyrush: I've added all the functionality I'd like to include with the first iteration of this capability, so this pull request is ready for review whenever you get a chance.

File: src/docs/sphinx/blueprint_mesh.rst [hunk 1]
```diff
@@ -443,10 +551,14 @@ supported values for this parameter and their corresponding effects are outlined
 +--------------------------------+--------------------+-------------------+-------------------+------------------+
 | `quads <Quads_>`_              | 2d                 | explicit          | explicit          | quad             |
 +--------------------------------+--------------------+-------------------+-------------------+------------------+
+| `polygons <Polygons_>`_        | 2d                 | explicit          | explicit          | polygon          |
++--------------------------------+--------------------+-------------------+-------------------+------------------+
 | `tets <Tets_>`_                | 3d                 | explicit          | explicit          | tet              |
 +--------------------------------+--------------------+-------------------+-------------------+------------------+
 | `hexs <Hexs_>`_                | 3d                 | explicit          | explicit          | hex              |
 +--------------------------------+--------------------+-------------------+-------------------+------------------+
+| `polyhedrons <Polyhedrons_>`_  | 3d                 | explicit          | explicit          | polyhedron       |
```
Comment (cyrush) [inline src/docs/sphinx/blueprint_mesh.rst]: polyhedra

File: src/docs/sphinx/blueprint_mesh.rst [hunk 2]
```diff
@@ -627,6 +764,32 @@ Hexs
 
     Pseudocolor plot of ``basic`` (element type 'hexs')
 
+Polyhedrons
```
Comment (cyrush) [inline src/docs/sphinx/blueprint_mesh.rst]: Polyhedra

File: src/libs/blueprint/conduit_blueprint_mesh_examples.cpp [hunk 3]
```diff
@@ -2087,30 +2118,46 @@ basic(const std::string &mesh_type,
 {
     // NOTE(JRC): The basic mesh example only supports simple, homogenous
     // element types that can be spanned by zone-centered fields.
-    const std::string mesh_types[7] = {
+    const std::string mesh_types[] = {
+        "uniform", "rectilinear", "structured",
+        "tris", "quads", "polygons",
+        "tets", "hexs", "polyhedrons"};
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_examples.cpp]: polyhedra

File: src/docs/sphinx/blueprint_mesh.rst (hunk 1 above)
Comment (rhornung67) [reply src/docs/sphinx/blueprint_mesh.rst]: polyhedroids?

Comment (cyrush) [reply src/docs/sphinx/blueprint_mesh.rst]: polyhadrons

**Pull request merged**